tests/unit/internal-session-test.js

```diff
@@ -189,6 +196,38 @@ describe('InternalSession', () => {
     describe('when the restored data does not contain an authenticator factory', () => {
       itDoesNotRestore();
     });
+
+    describe('with older synchronous stores (< v1.1.0)', function() {
+      describe('when restoring the session', function() {
+        describe('when the store resolves restoration', function() {
+          beforeEach(() => {
+            sinon.stub(store, 'restore').returns({ authenticated: { authenticator: 'authenticator' } });
+            sinon.stub(store, 'persist').returns();
+            sinon.stub(store, 'clear').returns();
+          });
+
+          it('is authenticated', () => {
+            return session.restore().then(() => {
+              expect(session.get('isAuthenticated')).to.be.true;
+            });
+          });
+        });
+
+        describe('when the store rejects restoration', function() {
```

Comment:
Hm, I see what you mean - the problem is though that when the store isn't cleared on invalidation and thus all session data remains persisted, then once the user reloads the page they would be authenticated again as the authenticator would probably restore the session successfully from the session data. Maybe we can come up with a nice solution over the weekend…